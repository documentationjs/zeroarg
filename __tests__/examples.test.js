var exec = require("child_process").exec;
var path = require("path");

test("add --help", done => {
  exec(
    path.join(__dirname, "..", "examples", "add.js") + " --help",
    (error, stdout, stderr) => {
      expect(stdout).toMatchSnapshot();
      done();
    }
  );
});

test("add run", done => {
  exec(
    path.join(__dirname, "..", "examples", "add.js") + " run 1 2",
    (error, stdout, stderr) => {
      expect(stdout).toMatchSnapshot();
      done();
    }
  );
});

test("options --help", done => {
  exec(
    path.join(__dirname, "..", "examples", "options.js") + " --help",
    (error, stdout, stderr) => {
      expect(stderr).toBeFalsy();
      expect(stdout).toMatchSnapshot();
      done();
    }
  );
});

test("options run", done => {
  exec(
    path.join(__dirname, "..", "examples", "options.js") + " run 1 -b 2",
    (error, stdout, stderr) => {
      expect(stderr).toBeFalsy();
      expect(stdout).toMatchSnapshot();
      done();
    }
  );
});

test("options bad option", done => {
  exec(
    path.join(__dirname, "..", "examples", "options.js") + " run 1 -c unknown",
    (error, stdout, stderr) => {
      expect(stderr).toBeTruthy();
      expect(stderr).toMatchSnapshot();
      done();
    }
  );
});

test("options good options", done => {
  exec(
    path.join(__dirname, "..", "examples", "options.js") +
      " run 1 -a 1 -b 2 -c waffles",
    (error, stdout, stderr) => {
      expect(stdout).toMatchSnapshot();
      done();
    }
  );
});

test("hello --help", done => {
  exec(
    path.join(__dirname, "..", "examples", "default.js") + " run",
    (error, stdout, stderr) => {
      expect(stdout).toMatchSnapshot();
      done();
    }
  );
});

test("hello default", done => {
  exec(
    path.join(__dirname, "..", "examples", "default.js") + " run",
    (error, stdout, stderr) => {
      expect(stdout).toMatchSnapshot();
      done();
    }
  );
});

test("hello --name", done => {
  exec(
    path.join(__dirname, "..", "examples", "default.js") + " run --name=Bill",
    (error, stdout, stderr) => {
      expect(stdout).toMatchSnapshot();
      done();
    }
  );
});
