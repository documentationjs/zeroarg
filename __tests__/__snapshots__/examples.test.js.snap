// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add --help 1`] = `
"Commands:
  run [numbers..]

Options:
  --help  Show help                                                    [boolean]


Add numbers together


"
`;

exports[`add run 1`] = `
"3
"
`;

exports[`hello --help 1`] = `
"Hello, John
"
`;

exports[`hello --name 1`] = `
"Hello, Bill
"
`;

exports[`hello default 1`] = `
"Hello, John
"
`;

exports[`options --help 1`] = `
"Commands:
  run [numbers]

Options:
  --help  Show help                                                    [boolean]
  -a      ...                                                           [string]
  -b      ...                                                           [number]
  -c      ...      [string] [choices: \\"wiffles\\", \\"waffles\\"] [default: \\"wiffles\\"]


Add numbers together


"
`;

exports[`options bad option 1`] = `
"/Users/tmcw/src/zeroarg/examples/options.js run [numbers]

Options:
  --help  Show help                                                    [boolean]
  -a      ...                                                           [string]
  -b      ...                                                           [number]
  -c      ...      [string] [choices: \\"wiffles\\", \\"waffles\\"] [default: \\"wiffles\\"]

Invalid values:
  Argument: c, Given: \\"unknown\\", Choices: \\"wiffles\\", \\"waffles\\"
"
`;

exports[`options good options 1`] = `
"1 '1' 2 'waffles'
"
`;

exports[`options run 1`] = `
"1 undefined 2 'wiffles'
"
`;
